packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -157,6 +157,7 @@ export const GridRootStyles = styled('div', {
       },
       [`& .${gridClasses.cell} > *`]: {
         overflow: 'visible !important',
+        whiteSpace: 'nowrap',
       },
```

Comment:
We know what to `whitespace: nowrap` because we assume a certain layout for the cells, but for custom rendered cells this might not work. There is no general solution to that though, let's just wait until someone has that problem.